src/main/java/easton/Storage.java

```diff
@@ -0,0 +1,61 @@
+package easton;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+public class Storage {
+
+    private final Path filePath;
+
+    Storage(String fileName) throws IOException {
+        String currentDirectory = System.getProperty("user.dir");
+        Path folder = Paths.get(currentDirectory, "data");
+        Path filePath = Paths.get(folder.toString(), fileName);
+
+        if (Files.notExists(folder)) {
+            Files.createDirectory(folder);
+        }
+
+        if (Files.notExists(filePath)) {
+            Files.createFile(filePath);
+        }
+
+        this.filePath = filePath;
+    }
+
+    public ArrayList<String> retrieve() throws IOException {
```

Comment:
Perhaps you could consider changing the naming of the functions in this class to make it clearer? You could name this function "retrieveRecord" instead of just "retrieve"?

Reply:
Made the changes! thank you